### PR TITLE
filters: add wrapContent

### PR DESCRIPTION
**Pull request by AlexanderYastrebov**
#0

### CODE REVIEW

File: docs/reference/filters.md [hunk 1]
```diff
@@ -465,6 +466,26 @@ Example:
...
+
+Examples:
+
+```
+* -> wrapContent("foo", "baz") -> inlineContent("bar") -> <shunt>
+```
+
+```
+// JSON array of 300 zeros
+* -> wrapContent("[", "0]") -> repeatContent("0, ", 297) -> <shunt>
```
Comment (szuecs) [inline docs/reference/filters.md:486]: These are 298 "0", no?

File: filters/builtin/builtin.go [hunk 2]
```diff
@@ -165,6 +165,7 @@ func Filters() []filters.Spec {
 		NewOriginMarkerSpec(),
 		diag.NewRandom(),
 		diag.NewRepeat(),
+		diag.NewWrap(),
```
Comment (szuecs) [inline filters/builtin/builtin.go:168]: NewContemtWrap maybe?

File: docs/reference/filters.md (hunk 1 above)
Comment (AlexanderYastrebov) [reply docs/reference/filters.md:486]: Yeah, the comment is wrong, should be `// JSON array of 100 zeros`

297 is the total length of repeated content, i.e 99 times the `"0, "`.

File: filters/builtin/builtin.go (hunk 2 above)
Comment (AlexanderYastrebov) [reply filters/builtin/builtin.go:168]: For some reason we have `diag.NewRandom()` for `randomContent` and `diag.NewRepeat()` for `repeatContent`

File: docs/reference/filters.md (hunk 1 above)
Comment (AlexanderYastrebov) [reply docs/reference/filters.md:486]: Fixed the comment

**Comment by AlexanderYastrebov** [comment]
@szuecs PTAL

File: filters/builtin/builtin.go (hunk 2 above)
Comment (AlexanderYastrebov) [reply filters/builtin/builtin.go:168]: So either we keep the discrepancy or I can rename it

File: docs/reference/filters.md [hunk 3]
```diff
@@ -465,6 +466,26 @@ Example:
...
 ```
 
+### wrapContent
+
+Add prefix and suffix to the response.
+
+Parameters:
+
+* suffix (string)
+* prefix (string)
```
Comment (szuecs) [inline docs/reference/filters.md:476]: maybe prefix should be before suffix in this list

File: filters/diag/diag.go [hunk 4]
```diff
@@ -259,6 +276,52 @@ func (r *repeatReader) Read(p []byte) (int, error) {
...
+	return &wrap{prefix: []byte(prefix), suffix: []byte(suffix)}, nil
+}
+
+func (w *wrap) Request(ctx filters.FilterContext) {}
+
+func (w *wrap) Response(ctx filters.FilterContext) {
+	rsp := ctx.Response()
+
+	if s := rsp.Header.Get("Content-Length"); s != "" {
+		if n, err := strconv.ParseInt(s, 10, 64); err == nil {
```
Comment (szuecs) [inline filters/diag/diag.go:305]: What if Content-Length is here "-1"?

File: filters/diag/wrap_test.go [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+	rsp, body := get(p.URL + "/foobarbaz")
+
+	assert.Equal(t, "foobarbaz", body)
+	assert.Equal(t, rsp.ContentLength, int64(9))
+	assert.Equal(t, rsp.Header.Get("Content-Length"), "9")
+
+	rsp, body = get(p.URL + "/chunked")
+
+	assert.Equal(t, "foohelloworldbaz", body)
```
Comment (szuecs) [inline filters/diag/wrap_test.go:64]: :+1:

File: filters/diag/wrap_test.go [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+	assert.Equal(t, rsp.ContentLength, int64(-1))
+	assert.Equal(t, rsp.Header.Get("Content-Length"), "")
```
Comment (szuecs) [inline filters/diag/wrap_test.go:66]: ah ok I see

**Comment by szuecs** [comment]
:+1:

File: docs/reference/filters.md (hunk 3 above)
Comment (AlexanderYastrebov) [reply docs/reference/filters.md:475]: True, fixed

File: filters/diag/wrap_test.go (hunk 6 above)
Comment (AlexanderYastrebov) [reply filters/diag/wrap_test.go]: Changed to key presence check

**Comment by szuecs** [comment]
:+1:

**Pull request merged**